File: psra/web/src/main/java/by/arhor/psra/WebApplication.java
package by.arhor.psra;

import by.arhor.psra.config.UserRoleTokenEnhancer;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.MessageSource;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;
import org.springframework.context.support.ResourceBundleMessageSource;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.security.oauth2.provider.token.DefaultTokenServices;
import org.springframework.security.oauth2.provider.token.TokenEnhancer;
import org.springframework.security.oauth2.provider.token.TokenStore;
import org.springframework.security.oauth2.provider.token.store.JwtAccessTokenConverter;
import org.springframework.security.oauth2.provider.token.store.JwtTokenStore;

@SpringBootApplication
public class WebApplication {

  public static void main(String[] args) {
    SpringApplication.run(WebApplication.class, args);
  }

  @Bean
  public MessageSource messageSource() {
    final var messageSource = new ResourceBundleMessageSource();
    messageSource.setBasename("messages");
    return messageSource;
  }

  @Bean
  public TokenStore tokenStore(JwtAccessTokenConverter converter) {
      return new JwtTokenStore(converter);
  }

  @Bean
  public JwtAccessTokenConverter accessTokenConverter() {
    final var converter = new JwtAccessTokenConverter();
    converter.setSigningKey("key");
    return converter;
  }

  @Bean
  public PasswordEncoder passwordEncoder() {
    return new BCryptPasswordEncoder(5);
  }

  @Bean
  @Primary
  public DefaultTokenServices tokenServices(TokenStore store) {
    final var defaultTokenServices = new DefaultTokenServices();
    defaultTokenServices.setTokenStore(store);
    return defaultTokenServices;
  }

  @Bean
  public TokenEnhancer tokenEnhancer() {
    return new UserRoleTokenEnhancer();
  }

}

File: psra/web/src/main/scala/by/arhor/psra/WebApplication.scala
package by.arhor.psra

import by.arhor.psra.config.UserRoleTokenEnhancer
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.MessageSource
import org.springframework.context.annotation.{Bean, Primary}
import org.springframework.context.support.ResourceBundleMessageSource
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.security.oauth2.provider.token.store.{JwtAccessTokenConverter, JwtTokenStore}
import org.springframework.security.oauth2.provider.token.{DefaultTokenServices, TokenEnhancer, TokenStore}

@SpringBootApplication
class WebApplication

object WebApplication extends App {

  SpringApplication.run(classOf[WebApplication])

  @Bean
  def messageSource(): MessageSource = {
    val messageSource = new ResourceBundleMessageSource()
    messageSource.setBasename("messages")
    messageSource
  }

  @Bean
  def tokenStore(converter: JwtAccessTokenConverter): TokenStore = new JwtTokenStore(converter)

  @Bean
  def accessTokenConverter(): JwtAccessTokenConverter = {
    val converter = new JwtAccessTokenConverter()
    converter.setSigningKey("key")
    converter
  }

  @Bean
  def passwordEncoder(): PasswordEncoder = new BCryptPasswordEncoder(5)

  @Bean
  @Primary
  def tokenServices(store: TokenStore): DefaultTokenServices = {
    val defaultTokenServices = new DefaultTokenServices()
    defaultTokenServices.setTokenStore(store)
    defaultTokenServices
  }

  @Bean
  def tokenEnhancer(): TokenEnhancer = new UserRoleTokenEnhancer()

}
